src/main/java/com/programmers/ticketparis/service/performance/PerformanceService.java

```diff
@@ -0,0 +1,64 @@
+package com.programmers.ticketparis.service.performance;
+
+import com.programmers.ticketparis.domain.performance.Performance;
+import com.programmers.ticketparis.dto.performance.request.PerformanceCreateRequest;
+import com.programmers.ticketparis.dto.performance.request.PerformanceUpdateRequest;
+import com.programmers.ticketparis.dto.performance.response.PerformanceResponse;
+import com.programmers.ticketparis.repository.performance.PerformanceRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+public class PerformanceService {
+    private final PerformanceRepository performanceRepository;
+
+    @Transactional
+    public void createPerformance(PerformanceCreateRequest createRequest) {
+        if (createRequest.getStartDate().isAfter(createRequest.getEndDate())) {
+            throw new IllegalArgumentException("공연 시작 날짜는 공연 종료 날짜 이후가 될 수 없습니다.");
+        }
+
+        Performance performance = createRequest.toEntity();
+        performanceRepository.save(performance);
+    }
+
+    @Transactional
+    public PerformanceResponse updatePerformance(Long id, PerformanceUpdateRequest updateRequest) {
+        if (updateRequest.getStartDate().isAfter(updateRequest.getEndDate())) {
+            throw new IllegalArgumentException("공연 시작 날짜는 공연 종료 날짜 이후가 될 수 없습니다.");
+        }
+        performanceRepository.update(id, updateRequest);
+        return findPerformanceById(id)
+                .orElseThrow(() -> new NoSuchElementException("수정할 공연 정보를 찾지 못했습니다."));
+    }
+
+    public Optional<PerformanceResponse> findPerformanceById(Long id) {
+        Performance performance = performanceRepository.findById(id)
+                .orElseThrow(() -> new NoSuchElementException("해당 공연 정보는 존재하지 않습니다."));
+
+        return Optional.ofNullable(PerformanceResponse.fromEntity(performance));
+    }
+
+    public List<PerformanceResponse> findPerformanceAll() {
+        List<Performance> performances = performanceRepository.findAll();
+        return performances.stream()
+                .map(PerformanceResponse::fromEntity)
+                .collect(Collectors.toList());
+    }
+
+
+    @Transactional
+    public void deletePerformance(Long id) {
+        if (!performanceRepository.findById(id).isPresent()) {
+            throw new NoSuchElementException("삭제하려는 공연을 찾지 못했습니다.");
+        }
```

Comment:
`p3`

```suggestion
       int result = performanceRepository.existsById(id);
       if(!result) {
           throw new PerformanceException(ExceptionRule, rejectedValue)
       }
```
현재 작성되어있는 findById의 경우 값을 찾아서 불러오는 역할의 메소드라고 생각합니다! existsById로 존재하는지 여부만 테스트하고 결과값으로 Boolean 형을 사용하는 것은 어떻게 생각하시나요?

Reply:
작업한 후 팀 컨벤션이 정해져서요. 서비스단에서 `delete`와 `update`로직에서 exists 메소드가 필요하여, Repository 단에서 `existsById`를 Boolean 형으로 만든 후, 서비스단에서 해당 내용을 체크하기 위해 `validatePerformanceExists`메소드를 별도로 빼서 만들었습니다.
적용: f64cdb380e287ee512801b9f547fc834fa48b4ee